src/main/java/uk/gov/hmcts/reform/pip/account/management/service/AccountService.java

```diff
@@ -503,4 +503,27 @@ public PiUser getUserById(UUID userId) {
             "User with supplied user id: %s could not be found", userId)));
     }
 
+    public AzureAccount retrieveUser(String issuerId) {
+        try {
+            Optional<PiUser> adminUser = userRepository.findByUserId(UUID.fromString(issuerId));
```

Comment:
There's a chance that this would return a non-PI aad account which would then error when calling azure.

 I think we should use a seperate repository method that only returns PI_AAD users